### PR TITLE
Markdown and URLs fixes 

### PR DESCRIPTION
**Pull request by agabrys**
Hello, I cleaned up documentation:

* use as much as possible markdowns to display text/urls/quotes instead of html tags
* fix dead and invalid urls
* set #issues as links
* remove ignored MarkdownTOC
* remove duplicated sections

### CODE REVIEW

**Comment by darkoperator** [comment]
can you resolve the conflicts?

**Comment by agabrys** [comment]
Done.

**Comment by darkoperator** [comment]
awesome! thanks for the PR

File: Best-Practices/Introduction.md [hunk 1]
```diff
@@ -20,5 +20,5 @@ One final note about these Best Practices: the perspective of these guidelines h
 - [Error Handling](Error-Handling.md)
 - [Performance](Performance.md)
 - [Security](Security.md)
-- [Language, Interop and .Net](Language,%20Interop%20and%20.Net.md)
-- [Metadata, Versioning, and Packaging](Metadata,%20Versioning,%20and%20Packaging.md)
+- [Language, Interop and .Net](Language,-Interop-and-.Net.md)
+- [Metadata, Versioning, and Packaging](Metadata,-Versioning,-and-Packaging.md)
```
Comment (Jaykul) [inline Best-Practices/Introduction.md]: This one's wrong, I had to remove the commas from the file name.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -20,21 +20,21 @@ The *PowerShell Best Practices* are always evolving, and continue to be edited a
...
+  * [Naming Conventions](Style-Guide/Naming-Conventions.md)
+* [Best Practices (Introduction)](Best-Practices/Introduction.md)
+  * [Naming Conventions](Best-Practices/Naming-Conventions.md)
+  * [Building Reusable Tools](Best-Practices/Building-Reusable-Tools.md)
+  * [Output and Formatting](Best-Practices/Output-and-Formatting.md)
+  * [Error Handling](Best-Practices/Error-Handling.md)
+  * [Performance](Best-Practices/Performance.md)
+  * [Security](Best-Practices/Security.md)
+  * [Language, Interop and .Net](Best-Practices/Language%2C-Interop-and-.Net.md)
+  * [Metadata, Versioning, and Packaging](Best-Practices/Metadata%2C-Versioning%2C-and-Packaging.md)
```
Comment (Jaykul) [inline CONTRIBUTING.md]: These two are also wrong -- the %2C is that comma I had to remove from the file names.

File: README.md [hunk 3]
```diff
@@ -30,21 +30,21 @@ Having said that, remember: the points in the Best Practices documents and the S
...
+  * [Naming Conventions](Style-Guide/Naming-Conventions.md)
+* [Best Practices (Introduction)](Best-Practices/Introduction.md)
+  * [Naming Conventions](Best-Practices/Naming-Conventions.md)
+  * [Building Reusable Tools](Best-Practices/Building-Reusable-Tools.md)
+  * [Output and Formatting](Best-Practices/Output-and-Formatting.md)
+  * [Error Handling](Best-Practices/Error-Handling.md)
+  * [Performance](Best-Practices/Performance.md)
+  * [Security](Best-Practices/Security.md)
+  * [Language, Interop and .Net](Best-Practices/Language%2C-Interop-and-.Net.md)
+  * [Metadata, Versioning, and Packaging](Best-Practices/Metadata%2C-Versioning%2C-and-Packaging.md)
```
Comment (Jaykul) [inline README.md]: The last two here are also wrong -- the %2C is that comma I had to remove from the file names.

**Comment by agabrys** [comment]
Thank you for your work. I found three more broken links.
I fixed them in https://github.com/PoshCode/PowerShellPracticeAndStyle/pull/93

**Pull request merged**